harness/harness.sh

```
@@ -156,19 +156,31 @@ while true; do
+
+  if [ $BUILD_EXIT -ne 0 ]; then
+    log_error "❌ 빌드 실패! 오류 내용을 분석하여 Claude에게 피드백을 전달합니다."
+    PROMPT="You are a Senior Architect. The build FAILED after Claude's changes. \n\n[BUILD ERROR]\n$BUILD_LOG \n\n[DIFF]\n$DIFF_FULL \n\nAnalyze the error and set phase to 'REPLANNING' with a specific 'feedback' for Claude to fix it. Output ONLY valid JSON."
+  else
+    log_success "✅ 빌드 성공! 코드 리뷰를 진행합니다."
+    PROMPT="You are a Senior Architect. The build was SUCCESSFUL. Review the following changes based on the goal: $(jq -r '.goal' "$STATE_FILE")\n\n[DIFF SUMMARY]\n$DIFF_STAT\n\n[FULL DIFF]\n$DIFF_FULL\n\nIf the code quality is good, set phase to 'COMPLETED'. If not, set to 'REPLANNING' with feedback. Output ONLY valid JSON."
+  fi
+
+  RESULT=$(call_gemini "$PROMPT")
+  if save_state_safely "$RESULT"; then
+...
```

Shellcheck (0.11.0)

[warning] This is interpreted as a command name ending with '.'. Double check syntax.

(SC2288)